src/open_inwoner/accounts/tests/test_eherkenning_session.py

```diff
@@ -0,0 +1,252 @@
+        request = self.make_request_with_session()
+        request.session[
+            "_auth_user_backend"
+        ] = "open_inwoner.accounts.backends.DigiDOIDCBackend"
+
+        context = EHerkenningSessionContext(request)
+        with self.assertRaises(ValueError) as ctx:
+            context.change_authenticated_user(kvk=self.user.kvk, vestiging=None)
+
+        self.assertEqual(
+            str(ctx.exception),
+            "The user in this session was not authenticated via the expected "
+            "EHerkenning backends: open_inwoner.accounts.backends."
+            "EHerkenningOIDCBackend, eherkenning.backends.eHerkenningBackend, "
+            "eherkenning.mock.backends.eHerkenningBackend",
+        )
+
+    def test_change_authenticated_user_raises_on_non_eherkenning_user(self):
+        request = self.make_request_with_session()
+        request.session[
+            "_auth_user_backend"
+        ] = EHerkenningSessionContext._expected_auth_backends()[0]
+        request.user = DigidUserFactory()
+
+        context = EHerkenningSessionContext(request)
+        with self.assertRaises(ValueError) as ctx:
+            context.change_authenticated_user(kvk=self.user.kvk, vestiging=None)
+
+        self.assertEqual(str(ctx.exception), "User is not an eHerkenning user")
+
+    def test_change_authenticated_user_raises_on_branch_restriction(self):
+        request = self.make_request_with_session()
+
+        context = EHerkenningSessionContext(request)
+        request.session[
+            "_auth_user_backend"
+        ] = EHerkenningSessionContext._expected_auth_backends()[0]
+        request.user = self.user
+        context._set_branch_restriction(True)
+        with self.assertRaises(ValueError) as ctx:
+            context.change_authenticated_user(kvk=self.user.kvk, vestiging=None)
+
+        self.assertEqual(
+            str(ctx.exception),
+            "This session is branch restricted: you cannot switch to a "
+            "different eherkenning user",
+        )
+
+    def test_change_authenticated_user_raises_on_target_user_with_different_kvk(self):
+        request = self.make_request_with_session()
+        request.session[
+            "_auth_user_backend"
+        ] = EHerkenningSessionContext._expected_auth_backends()[0]
+        request.user = self.user
+        different_kvk_user = eHerkenningUserFactory(kvk="38804926")
+
+        context = EHerkenningSessionContext(request)
+        with self.assertRaises(ValueError) as ctx:
+            context.change_authenticated_user(
+                kvk=different_kvk_user.kvk, vestiging=None
+            )
+
+        self.assertEqual(
+            str(ctx.exception),
+            "You cannot change the authenticated user to the rechtspersoon or "
+            "vestiging of a different kvk number",
+        )
+        self.assertEqual(request.user, self.user)
+
+    def test_change_authenticated_user_with_branch_restriction_raises(
```

Comment:
Duplicate of l.107 ff. I like the formatting here better (separation of setup, test, assert with newlines), otherwise they're they same.